### ISSUE TITLE
MKLDNN doesn't link MKL and Intel OMP during PyTorch build

### ISSUE THREAD

**Issue by h6197627** [pytorch/pytorch#13544]
## 🐛 Bug

MKLDNN shared library built as a part of IDEEP interface does not link Intel MKL and Intel OMP. This happens because PyTorch's cmake/Modules/FindMKL.cmake script sets MKL_cmake_included variable to true and when it comes to configuring MKLDNN in third_party/ideep/mkl-dnn/cmake/MKL.cmake script it returns from the first line
```
if(MKL_cmake_included)
    return()
endif()
```
However, according to MKLDNN's MKL configuration script it should be linked with other libraries than PyTorch: MKLDNN uses either downloaded MKL-small or mkl_rt and PyTorch uses mkl_intel_lp64, mkl_gnu_thread,  mkl_core. So probably due to misconfiguration it is not linked with MKL and Intel OMP (it uses libgomp.so insted).

Before #12170 it was possible to provide system MKLDNN, but now unfortunately I can't do it. Maybe there is any way to still do it? Also it forces users to run outdated MKLDNN 0.14 version and it is possible to compile 0.16 with various speed improvements (and correct linkage).

## To Reproduce

> USE_MKLDNN=1 python3 setup.py build
> ldd build/lib/libmkldnn.so

## Expected behavior

libmkldnn.so should be linked with Intel MKL and Intel OMP libraries.

## Environment
 - PyTorch Version (e.g., 1.0): 1.0.0a0+0479517
 - OS (e.g., Linux): Ubuntu 18.04
 - How you installed PyTorch (`conda`, `pip`, source): source
 - Build command you used (if compiling from source): USE_MKLDNN=1 python3 setup.py build
 - Python version: 3.6.6

**Comment by h6197627** [comment]
OK, it seems that MKL linking was fixed with #13449
Last question before closing, @{gujinghui}, could you explain why FindMKL.cmake prefer gnu threads for GCC compiler
```
IF(CMAKE_COMPILER_IS_GNUCC)
  SET(mklthreads "mkl_gnu_thread" "mkl_intel_thread")
  SET(mklifaces  "intel" "gf")
  SET(mklrtls "gomp" "iomp5")
ELSE(CMAKE_COMPILER_IS_GNUCC)
```
mkl_intel_thread and iomp5 goes as a 2nd option which won't be activated if mkl_gnu_thread and gomp checking successfull.

**Comment by h6197627** [comment]
OK, it seems that MKL linking was fixed with #13449
Last question before closing, @gujinghui, could you explain why FindMKL.cmake prefer gnu threads for GCC compiler
```
IF(CMAKE_COMPILER_IS_GNUCC)
  SET(mklthreads "mkl_gnu_thread" "mkl_intel_thread")
  SET(mklifaces  "intel" "gf")
  SET(mklrtls "gomp" "iomp5")
ELSE(CMAKE_COMPILER_IS_GNUCC)
```
mkl_intel_thread and iomp5 goes as a 2nd option which won't be activated if mkl_gnu_thread and gomp checking successfull.

**Comment by gujinghui** [comment]
Perhaps @yinghai can give more details on omp issue.
